doc/security/_index.md

```diff
@@ -0,0 +1,111 @@
+---
+title: "OpenTitan Security Introduction"
```

Comment:
"Introduction" or "Overview"?

Reply:
"OpenTitan Security Overview" sounds better. Updated the title.